File: benchmarks/bubble.rb
class Array
  def bubblesort1!
    length.times do |j|
      for i in 1...(length - j)
        if self[i] < self[i - 1]
          self[i], self[i - 1] = self[i - 1], self[i]
        end
      end
    end
    self
  end
end

start = Time.now
ary = [5, 6, 1, 2, 9, 14, 2, 15, 6, 7, 8, 97]
ary.bubblesort1!
p ary
puts "elapsed: " + (Time.now - start).to_s

File: benchmarks/bubble.py
import time

def shortBubbleSort(alist):
    exchanges = True
    passnum = len(alist)-1
    while passnum > 0 and exchanges:
       exchanges = False
       for i in range(passnum):
           if alist[i]>alist[i+1]:
               exchanges = True
               temp = alist[i]
               alist[i] = alist[i+1]
               alist[i+1] = temp
       passnum = passnum-1

alist = [5, 6, 1, 2, 9, 14, 2, 15, 6, 7, 8, 97]
start = time.time()
shortBubbleSort(alist)
print(alist)
end = time.time()
elapsed = end - start
print("Elapsed time %.5f" % (elapsed))
